onadata/apps/api/viewsets/v2/tableau_viewset.py

```diff
@@ -188,39 +205,74 @@ def data(self, request, **kwargs):
                 return Response(status=status.HTTP_404_NOT_FOUND)
 
             xform = self.object.content_object
-            if xform.is_merged_dataset:
-                qs_kwargs = {
-                    "xform_id__in": list(
+
+            if should_paginate or count:
+                qs_kwargs = {}
+
+                if xform.is_merged_dataset:
+                    xform_pks = list(
                         xform.mergedxform.xforms.values_list("pk", flat=True)
                     )
-                }
-            else:
-                qs_kwargs = {"xform_id": xform.pk}
+                    qs_kwargs = {"xform__pk__in": xform_pks}
+
+                else:
+                    qs_kwargs = {"xform__pk": xform.pk}
+
+                if gt_id:
+                    qs_kwargs.update({"id__gt": gt_id})
+
+                self.data_count = (
+                    Instance.objects.filter(**qs_kwargs, deleted_at__isnull=True)
+                    .only("pk")
+                    .count()
```

Comment:
Lets avoid doing an extra count query and use `xform.num_of_submissions` instead

Reply:
@KipSigei That would work, but we are applying filters so `xform.num_of_submissions` will not give us the correct result